DependencyInjection/Compiler/LoggerChannelPass.php

```diff
@@ -112,4 +112,22 @@ protected function createLogger($channel, $loggerId, ContainerBuilder $container
             $this->channels[] = $channel;
         }
     }
+
+    /**
+     * Creates a copy of a reference and alters the service ID.
+     *
+     * @param Reference $reference
+     * @param string    $serviceId
+     *
+     * @return Reference
+     */
+    private function changeReference(Reference $reference, $serviceId)
+    {
+        if (method_exists($reference, 'isStrict')) {
+            // Stay compatible with Symfony 2
+            return new Reference($serviceId, $reference->getInvalidBehavior(), $reference->isStrict());
```

Comment:
This still triggers alot of deprecation warnings when warming the cache in the standard edition. We need a different solution. We could just use `$reference->isStrict(false)` to prevent the deprecation warning. Or can we not ignore the `strict` flag completely? It's rather unlikely to have a scope problem with a logger...